File: 2575. Find the Divisibility Array of a String/Solution.java
class Solution {
    public int[] divisibilityArray(String word, int m) {
        int n = word.length();
        long r = 0;
        int[] res = new int[n];
        for (int i = 0; i < n; i++) {
            r = (r * 10 + word.charAt(i) - '0') % m;
            if (r == 0) res[i] = 1;
        }
        return res;
    }
}

File: 2575. Find the Divisibility Array of a String/Solution.py
class Solution:
    def divisibilityArray(self, word: str, m: int) -> List[int]:
        n, r = len(word), 0
        res = [0 for i in range(n)]
        for i in range(n):
            r = (r * 10 + int(word[i])) % m
            if r == 0: res[i] = 1
        return res